bin/lib/common.sh

```diff
@@ -0,0 +1,318 @@
+#!/bin/bash
+
+# shellcheck disable=SC2181
+# -- https://www.shellcheck.net/wiki/SC2181
+
+set -eu
+
+declare DEBUG=${DEBUG:-0}
+declare PROJECT_DIR; PROJECT_DIR="$(git rev-parse --show-toplevel 2> /dev/null)" || \
+    (echo_err "$(basename "${0}") commands use git to set directory context. Exiting..." && exit 135)
+if [[ -z "${PROJECT_DIR}" ]]; then
+  echo_err "$(basename "${0}") commands should be run from the directory context of the OSMT git repo. Exiting..."
+  exit 135
+fi
+
+# new line formatted to indent with echo_err / echo_info etc
+declare -r INDENT="       "
+declare -r NL="\n${INDENT}"
+
+source_env_file() {
+  local env_file="${1}"
+
+  if [[ ! -f "${env_file}" ||  ! -r "${env_file}" ]]; then
+    echo_err "Can not access ${env_file}. You can initialize the environment files by running $(basename "${0}") -i$"
+    return 1
+  fi
+
+  _validate_env_file "${env_file}" || return 1
+
+  echo_info "Sourcing ${env_file}"
+  set -o allexport
+  # shellcheck source=/dev/null
+  source "${env_file}"
+  set +o allexport
+}
+
+source_osmt_env_file() {
```

Comment:
The name of this function seems to be confusing with "source_env_file" which does the actual sourcing.

Reply:
I agree the names are a bit confusing but it seems the intent of `source_env_file()` is to do the actual sourcing of any env file (as you've indicated), but that this `source_osmt_env_file()` is to specifically handle instances when certain environment variables already exist in the environment - such as when running in the GitHub containers as the variables are injected externally. So we want to make sure OSMT envs exist and then source the actual env files if the envs aren't already present.

This PR's intent is to refactor the functionality to maximize utility between the existing osmt_cli.sh, the new api test scripts, and anything else going forward.

Reply:
it can be named as "bypass_env_file_source" which describes the actual functionality.